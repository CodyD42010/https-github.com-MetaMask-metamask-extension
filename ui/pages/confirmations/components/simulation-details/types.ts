import { Hex } from '@metamask/utils';
import { TokenStandard } from '../../../../../shared/constants/transaction';
import { Numeric } from '../../../../../shared/modules/Numeric';

export const NATIVE_ASSET_IDENTIFIER: NativeAssetIdentifier = {
  standard: TokenStandard.none,
};

/**
 * Describes an amount of fiat.
 */
export const FIAT_UNAVAILABLE = null;
export type FiatAmountAvailable = number;
export type FiatAmount = FiatAmountAvailable | typeof FIAT_UNAVAILABLE;

/**
 * Identifies the native asset of a chain.
 */
export type NativeAssetIdentifier = {
  standard: TokenStandard.none;
  address?: undefined;
  tokenId?: undefined;
};

/**
 * Uniquely identifies a token asset on a chain.
 */
export type TokenAssetIdentifier = {
  standard: Exclude<TokenStandard, TokenStandard.none>;
  address: Hex;
  tokenId?: Hex;
};

export type AssetIdentifier = Readonly<
  NativeAssetIdentifier | TokenAssetIdentifier
>;

/**
 * Represents an amount of an asset, including its magnitude and sign.
 */
export type Amount = Readonly<{
  /**
   * Indicates whether the amount is negative (e.g., a decrease in balance).
   */
  isNegative: boolean;

  /**
   * The quantity of the smallest denomination of the asset (base units),
   * represented as a hexadecimal string.
   * For example: In the case of ETH, this would be the number of wei.
   */
  quantity: Hex;

  /**
   * The number of decimal places the associated asset supports.
   *
   * This value is the negation of the exponent used when converting
   * the quantity to the decimal amount of a token.
   *
   * To calculate the token amount in decimal form, use the formula:
   * `tokenAmount = hexToDecimal(quantity) / (10 ^ decimals)`
   *
   * Example: If the asset is ETH, the quantity is expressed in wei
   * (the smallest unit of ETH) and decimals would be 18. The amount
   * of ETH tokens would be: `ethAmount = quantity / (10 ^ 18)`
   */
  decimals: number;

  /**
   * The numeric representation of the amount, taking into account the
   * sign, quantity and decimals.
   */
  numeric: Numeric;
}>;

/**
 * Describes a change in an asset's balance to a user's wallet.
 */
export type BalanceChange = {
  /**
   * The asset identifier for the balance change.
   */
  asset: AssetIdentifier;
  /**
   * The amount of the asset that changed.
   */
  amount: Amount;
  /**
   * The amount of fiat currency that corresponds to the asset amount.
   */
  fiatAmount: FiatAmount;
};
