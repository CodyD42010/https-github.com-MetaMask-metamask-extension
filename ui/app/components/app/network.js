const Component = require('react').Component
const PropTypes = require('prop-types')
const h = require('react-hyperscript')
const connect = require('react-redux').connect
const classnames = require('classnames')
const inherits = require('util').inherits
const NetworkDropdownIcon = require('./dropdowns/components/network-dropdown-icon')

Network.contextTypes = {
  t: PropTypes.func,
}

module.exports = connect()(Network)


inherits(Network, Component)

function Network () {
  Component.call(this)
}

Network.prototype.render = function () {
  const props = this.props
  const context = this.context
  const networkNumber = props.network
  let providerName, providerNick, providerUrl
  try {
    providerName = props.provider.type
    providerNick = props.provider.nickname || ''
    providerUrl = props.provider.rpcTarget
  } catch (e) {
    providerName = null
  }
  const providerId = providerNick || providerName || providerUrl || null
  let iconName
  let hoverText

  if (providerName === 'mainnet') {
    hoverText = context.t('mainnet')
    iconName = 'ethereum-network'
  } else if (providerName === 'ropsten') {
    hoverText = context.t('ropsten')
    iconName = 'ropsten-test-network'
  } else if (providerName === 'kovan') {
    hoverText = context.t('kovan')
    iconName = 'kovan-test-network'
  } else if (providerName === 'rinkeby') {
    hoverText = context.t('rinkeby')
    iconName = 'rinkeby-test-network'
  } else if (providerName === 'goerli') {
    hoverText = context.t('goerli')
    iconName = 'goerli-test-network'
  } else {
    hoverText = providerId
    iconName = 'private-network'
  }

  return (
    h('div.network-component.pointer', {
      className: classnames({
        'network-component--disabled': this.props.disabled,
        'ethereum-network': providerName === 'mainnet',
        'ropsten-test-network': providerName === 'ropsten',
        'kovan-test-network': providerName === 'kovan',
        'rinkeby-test-network': providerName === 'rinkeby',
        'goerli-test-network': providerName === 'goerli',
      }),
      title: hoverText,
      onClick: (event) => {
        if (!this.props.disabled) {
          this.props.onClick(event)
        }
      },
    }, [
      (function () {
        switch (iconName) {
          case 'ethereum-network':
            return h('.network-indicator', [
              h(NetworkDropdownIcon, {
                backgroundColor: '#038789', // $blue-lagoon
                nonSelectBackgroundColor: '#15afb2',
                loading: networkNumber === 'loading',
              }),
              h('.network-name', context.t('mainnet')),
              h('.network-indicator__down-arrow'),
            ])
          case 'ropsten-test-network':
            return h('.network-indicator', [
              h(NetworkDropdownIcon, {
                backgroundColor: '#e91550', // $crimson
                nonSelectBackgroundColor: '#ec2c50',
                loading: networkNumber === 'loading',
              }),
              h('.network-name', context.t('ropsten')),
              h('.network-indicator__down-arrow'),
            ])
          case 'kovan-test-network':
            return h('.network-indicator', [
              h(NetworkDropdownIcon, {
                backgroundColor: '#690496', // $purple
                nonSelectBackgroundColor: '#b039f3',
                loading: networkNumber === 'loading',
              }),
              h('.network-name', context.t('kovan')),
              h('.network-indicator__down-arrow'),
            ])
          case 'rinkeby-test-network':
            return h('.network-indicator', [
              h(NetworkDropdownIcon, {
                backgroundColor: '#ebb33f', // $tulip-tree
                nonSelectBackgroundColor: '#ecb23e',
                loading: networkNumber === 'loading',
              }),
              h('.network-name', context.t('rinkeby')),
              h('.network-indicator__down-arrow'),
            ])
          case 'goerli-test-network':
            return h('.network-indicator', [
              h(NetworkDropdownIcon, {
                backgroundColor: '#3099f2', // $dodger-blue
                nonSelectBackgroundColor: '#ecb23e',
                loading: networkNumber === 'loading',
              }),
              h('.network-name', context.t('goerli')),
              h('.network-indicator__down-arrow'),
            ])
          default:
            return h('.network-indicator', [
              networkNumber === 'loading'
                ? h('span.pointer.network-loading-spinner', {
                  onClick: (event) => this.props.onClick(event),
                }, [
                  h('img', {
                    title: context.t('attemptingConnect'),
                    src: 'images/loading.svg',
                  }),
                ])
                : h('i.fa.fa-question-circle.fa-lg', {
                  style: {
                    color: 'rgb(125, 128, 130)',
                  },
                }),

              h('.network-name', providerName === 'localhost' ? context.t('localhost') : providerNick || context.t('privateNetwork')),
              h('.network-indicator__down-arrow'),
            ])
        }
      })(),
    ])
  )
}
