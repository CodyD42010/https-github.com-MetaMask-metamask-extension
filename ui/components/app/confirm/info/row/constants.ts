export const TEST_ADDRESS = '0x5CfE73b6021E818B776b421B1c4Db2474086a7e1';
