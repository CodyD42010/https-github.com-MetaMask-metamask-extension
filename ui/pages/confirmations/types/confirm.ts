import { ApprovalControllerState } from '@metamask/approval-controller';
import { TransactionType } from '@metamask/transaction-controller';

export type SignatureRequestType = {
  chainId?: string;
  id: string;
  msgParams?: {
    from: string;
    origin: string;
    data: string;
  };
  type: TransactionType;
  custodyId?: string;
};

export type Confirmation = SignatureRequestType;

export type ConfirmMetamaskState = {
  confirm: {
    currentConfirmation?: Confirmation;
    isScrollToBottomNeeded?: boolean;
  };
  metamask: {
    pendingApprovals: ApprovalControllerState['pendingApprovals'];
    approvalFlows: ApprovalControllerState['approvalFlows'];
  };
};
