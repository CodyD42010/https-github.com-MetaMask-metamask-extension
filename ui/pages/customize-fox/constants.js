export const POLISH_OPTIONS = [
  {
    value: 'generative',
    name: 'Generative',
  },
  {
    value: 'ai',
    name: 'AI',
  },
  {
    value: 'editorSelection',
    name: "Editor's selection",
  },
  {
    value: 'default',
    name: 'Your favourite fox',
  },
];

export const EditorSelectionOptions = [
  {
    value: '1',
    name: 'Rose Hue',
  },
  {
    value: '2',
    name: 'Cocoa Chic',
  },
  {
    value: '3',
    name: 'Chromatic Joy',
  },
  {
    value: '4',
    name: 'Woodland Whisper',
  },
  {
    value: '5',
    name: 'Deep Blush',
  },
];
