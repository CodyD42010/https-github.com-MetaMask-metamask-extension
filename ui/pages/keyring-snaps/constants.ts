export const METAMASK_DEVELOPER = 'metamask';
