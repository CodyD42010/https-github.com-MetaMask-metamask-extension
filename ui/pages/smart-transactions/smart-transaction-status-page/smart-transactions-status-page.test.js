import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { SmartTransactionStatuses } from '@metamask/smart-transactions-controller/dist/types';

import {
  renderWithProvider,
  createSwapsMockStore,
} from '../../../../test/jest';
import { CHAIN_IDS } from '../../../../shared/constants/network';
import { SmartTransactionStatusPage } from '.';

const middleware = [thunk];

describe('SmartTransactionStatusPage', () => {
  const requestState = {
    smartTransaction: {
      status: SmartTransactionStatuses.PENDING,
      creationTime: Date.now(),
    },
  };

  it('renders the component with initial props', () => {
    const store = configureMockStore(middleware)(createSwapsMockStore());
    const { getByText } = renderWithProvider(
      <SmartTransactionStatusPage requestState={requestState} />,
      store,
    );
    expect(getByText('Submitting your transaction')).toBeInTheDocument();
  });

  it('renders the "Sorry for the wait" pending status', () => {
    const store = configureMockStore(middleware)(createSwapsMockStore());
    const newRequestState = {
      ...requestState,
      smartTransaction: {
        ...requestState.smartTransaction,
        creationTime: 1519211809934,
      },
    };
    const { getByText } = renderWithProvider(
      <SmartTransactionStatusPage requestState={newRequestState} />,
      store,
    );
    expect(getByText('Sorry for the wait')).toBeInTheDocument();
  });

  it('renders the "success" STX status', () => {
    const mockStore = createSwapsMockStore();
    const latestSmartTransaction =
      mockStore.metamask.smartTransactionsState.smartTransactions[
        CHAIN_IDS.MAINNET
      ][1];
    latestSmartTransaction.status = SmartTransactionStatuses.SUCCESS;
    requestState.smartTransaction = latestSmartTransaction;
    const store = configureMockStore(middleware)(mockStore);
    const { getByText } = renderWithProvider(
      <SmartTransactionStatusPage requestState={requestState} />,
      store,
    );
    expect(getByText('Your transaction is complete')).toBeInTheDocument();
  });

  it('renders the "reverted" STX status', () => {
    const mockStore = createSwapsMockStore();
    const latestSmartTransaction =
      mockStore.metamask.smartTransactionsState.smartTransactions[
        CHAIN_IDS.MAINNET
      ][1];
    latestSmartTransaction.status = SmartTransactionStatuses.REVERTED;
    requestState.smartTransaction = latestSmartTransaction;
    const store = configureMockStore(middleware)(mockStore);
    const { getByText } = renderWithProvider(
      <SmartTransactionStatusPage requestState={requestState} />,
      store,
    );
    expect(getByText('Your transaction failed')).toBeInTheDocument();
    expect(
      getByText(
        'Sudden market changes can cause failures. If the problem continues, reach out to MetaMask customer support.',
      ),
    ).toBeInTheDocument();
  });

  it('renders the "cancelled" STX status', () => {
    const mockStore = createSwapsMockStore();
    const latestSmartTransaction =
      mockStore.metamask.smartTransactionsState.smartTransactions[
        CHAIN_IDS.MAINNET
      ][1];
    requestState.smartTransaction = latestSmartTransaction;
    latestSmartTransaction.status = SmartTransactionStatuses.CANCELLED;
    const store = configureMockStore(middleware)(mockStore);
    const { getByText } = renderWithProvider(
      <SmartTransactionStatusPage requestState={requestState} />,
      store,
    );
    expect(getByText('Your transaction was canceled')).toBeInTheDocument();
    expect(
      getByText(
        `Your transaction couldn't be completed, so it was canceled to save you from paying unnecessary gas fees.`,
      ),
    ).toBeInTheDocument();
  });

  it('renders the "deadline_missed" STX status', () => {
    const mockStore = createSwapsMockStore();
    const latestSmartTransaction =
      mockStore.metamask.smartTransactionsState.smartTransactions[
        CHAIN_IDS.MAINNET
      ][1];
    latestSmartTransaction.status =
      SmartTransactionStatuses.CANCELLED_DEADLINE_MISSED;
    requestState.smartTransaction = latestSmartTransaction;
    const store = configureMockStore(middleware)(mockStore);
    const { getByText } = renderWithProvider(
      <SmartTransactionStatusPage requestState={requestState} />,
      store,
    );
    expect(getByText('Your transaction was canceled')).toBeInTheDocument();
  });

  it('renders the "unknown" STX status', () => {
    const mockStore = createSwapsMockStore();
    const latestSmartTransaction =
      mockStore.metamask.smartTransactionsState.smartTransactions[
        CHAIN_IDS.MAINNET
      ][1];
    latestSmartTransaction.status = SmartTransactionStatuses.UNKNOWN;
    requestState.smartTransaction = latestSmartTransaction;
    const store = configureMockStore(middleware)(mockStore);
    const { getByText } = renderWithProvider(
      <SmartTransactionStatusPage requestState={requestState} />,
      store,
    );
    expect(getByText('Your transaction failed')).toBeInTheDocument();
  });
});
