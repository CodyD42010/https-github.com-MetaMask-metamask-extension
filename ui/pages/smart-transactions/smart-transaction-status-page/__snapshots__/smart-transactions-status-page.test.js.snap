// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmartTransactionStatusPage renders the "Sorry for the wait" pending status 1`] = `
<div>
  <div
    class="mm-box smart-transaction-status-page mm-box--margin-bottom-0 mm-box--display-flex mm-box--flex-direction-column mm-box--align-items-center mm-box--width-full mm-box--height-full mm-box--border-style-none"
  >
    <div
      class="mm-box mm-box--padding-right-4 mm-box--padding-left-4 mm-box--display-flex mm-box--flex-direction-column mm-box--justify-content-center mm-box--align-items-center mm-box--width-full"
      style="flex-grow: 1;"
    >
      <div
        class="mm-box mm-box--padding-right-6 mm-box--padding-left-6 mm-box--display-flex mm-box--flex-direction-column mm-box--align-items-center"
      >
        <div
          class="mm-box mm-box--display-flex"
          style="font-size: 48px;"
        >
          <span
            class="mm-box mm-icon mm-icon--size-inherit mm-box--margin-bottom-4 mm-box--display-inline-block mm-box--color-primary-default"
            style="mask-image: url('./images/icons/clock.svg');"
          />
        </div>
        <h4
          class="mm-box mm-text mm-text--heading-md mm-text--font-weight-bold mm-box--color-text-default"
        >
          Sorry for the wait
        </h4>
        <div
          class="mm-box mm-box--display-flex mm-box--flex-direction-column mm-box--align-items-center mm-box--width-full"
        >
          <div
            class="smart-transaction-status-page__loading-bar-container"
          >
            <div
              class="smart-transaction-status-page__loading-bar"
              style="width: 100%;"
            />
          </div>
        </div>
        <div
          class="mm-box mm-box--display-flex mm-box--flex-direction-column mm-box--align-items-center"
        >
          <p
            class="mm-box mm-text mm-text--body-sm mm-box--margin-top-2 mm-box--color-text-alternative"
          >
            <span>
               
              If your transaction is not finalized within 
              <p
                class="mm-box mm-text smart-transaction-status-page__countdown mm-text--body-sm mm-text--text-align-center mm-box--display-inline-block mm-box--color-text-alternative"
              >
                0:00
              </p>
              , it will be canceled and you will not be charged for gas.
               
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      class="mm-box smart-transaction-status-page__footer mm-box--padding-4 mm-box--padding-bottom-0 mm-box--display-flex mm-box--flex-direction-column mm-box--width-full"
    >
      <button
        class="mm-box mm-text mm-button-base mm-button-base--size-md mm-button-secondary mm-text--body-md-medium mm-box--margin-top-3 mm-box--padding-0 mm-box--padding-right-4 mm-box--padding-left-4 mm-box--display-inline-flex mm-box--justify-content-center mm-box--align-items-center mm-box--width-full mm-box--color-primary-default mm-box--background-color-transparent mm-box--rounded-pill mm-box--border-color-primary-default box--border-style-solid box--border-width-1"
        data-testid="smart-transaction-status-page-footer-close-button"
      >
        View activity
      </button>
    </div>
  </div>
</div>
`;

exports[`SmartTransactionStatusPage renders the "cancelled" STX status 1`] = `
<div>
  <div
    class="mm-box smart-transaction-status-page mm-box--margin-bottom-0 mm-box--display-flex mm-box--flex-direction-column mm-box--align-items-center mm-box--width-full mm-box--height-full mm-box--border-style-none"
  >
    <div
      class="mm-box mm-box--padding-right-4 mm-box--padding-left-4 mm-box--display-flex mm-box--flex-direction-column mm-box--justify-content-center mm-box--align-items-center mm-box--width-full"
      style="flex-grow: 1;"
    >
      <div
        class="mm-box mm-box--padding-right-6 mm-box--padding-left-6 mm-box--display-flex mm-box--flex-direction-column mm-box--align-items-center"
      >
        <div
          class="mm-box mm-box--display-flex"
          style="font-size: 48px;"
        >
          <span
            class="mm-box mm-icon mm-icon--size-inherit mm-box--margin-bottom-4 mm-box--display-inline-block mm-box--color-error-default"
            style="mask-image: url('./images/icons/danger.svg');"
          />
        </div>
        <h4
          class="mm-box mm-text mm-text--heading-md mm-text--font-weight-bold mm-box--color-text-default"
        >
          Your transaction was canceled
        </h4>
        <div
          class="mm-box mm-box--display-flex mm-box--flex-direction-column mm-box--align-items-center"
        >
          <p
            class="mm-box mm-text mm-text--body-sm mm-box--margin-top-2 mm-box--color-text-alternative"
          >
            Your transaction couldn't be completed, so it was canceled to save you from paying unnecessary gas fees.
          </p>
        </div>
        <div
          class="mm-box mm-box--margin-top-2 mm-box--display-flex mm-box--flex-direction-column"
        >
          <button
            class="mm-box mm-text mm-button-base mm-button-link mm-button-link--size-auto mm-text--body-md-medium mm-box--padding-0 mm-box--padding-right-0 mm-box--padding-left-0 mm-box--display-inline-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-primary-default mm-box--background-color-transparent"
            type="link"
          >
            View  transaction
          </button>
        </div>
      </div>
    </div>
    <div
      class="mm-box smart-transaction-status-page__footer mm-box--padding-4 mm-box--padding-bottom-0 mm-box--display-flex mm-box--flex-direction-column mm-box--width-full"
    >
      <button
        class="mm-box mm-text mm-button-base mm-button-base--size-md mm-button-secondary mm-text--body-md-medium mm-box--margin-top-3 mm-box--padding-0 mm-box--padding-right-4 mm-box--padding-left-4 mm-box--display-inline-flex mm-box--justify-content-center mm-box--align-items-center mm-box--width-full mm-box--color-primary-default mm-box--background-color-transparent mm-box--rounded-pill mm-box--border-color-primary-default box--border-style-solid box--border-width-1"
        data-testid="smart-transaction-status-page-footer-close-button"
      >
        View activity
      </button>
    </div>
  </div>
</div>
`;

exports[`SmartTransactionStatusPage renders the "deadline_missed" STX status 1`] = `
<div>
  <div
    class="mm-box smart-transaction-status-page mm-box--margin-bottom-0 mm-box--display-flex mm-box--flex-direction-column mm-box--align-items-center mm-box--width-full mm-box--height-full mm-box--border-style-none"
  >
    <div
      class="mm-box mm-box--padding-right-4 mm-box--padding-left-4 mm-box--display-flex mm-box--flex-direction-column mm-box--justify-content-center mm-box--align-items-center mm-box--width-full"
      style="flex-grow: 1;"
    >
      <div
        class="mm-box mm-box--padding-right-6 mm-box--padding-left-6 mm-box--display-flex mm-box--flex-direction-column mm-box--align-items-center"
      >
        <div
          class="mm-box mm-box--display-flex"
          style="font-size: 48px;"
        >
          <span
            class="mm-box mm-icon mm-icon--size-inherit mm-box--margin-bottom-4 mm-box--display-inline-block mm-box--color-error-default"
            style="mask-image: url('./images/icons/danger.svg');"
          />
        </div>
        <h4
          class="mm-box mm-text mm-text--heading-md mm-text--font-weight-bold mm-box--color-text-default"
        >
          Your transaction was canceled
        </h4>
        <div
          class="mm-box mm-box--display-flex mm-box--flex-direction-column mm-box--align-items-center"
        >
          <p
            class="mm-box mm-text mm-text--body-sm mm-box--margin-top-2 mm-box--color-text-alternative"
          >
            Your transaction couldn't be completed, so it was canceled to save you from paying unnecessary gas fees.
          </p>
        </div>
        <div
          class="mm-box mm-box--margin-top-2 mm-box--display-flex mm-box--flex-direction-column"
        >
          <button
            class="mm-box mm-text mm-button-base mm-button-link mm-button-link--size-auto mm-text--body-md-medium mm-box--padding-0 mm-box--padding-right-0 mm-box--padding-left-0 mm-box--display-inline-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-primary-default mm-box--background-color-transparent"
            type="link"
          >
            View  transaction
          </button>
        </div>
      </div>
    </div>
    <div
      class="mm-box smart-transaction-status-page__footer mm-box--padding-4 mm-box--padding-bottom-0 mm-box--display-flex mm-box--flex-direction-column mm-box--width-full"
    >
      <button
        class="mm-box mm-text mm-button-base mm-button-base--size-md mm-button-secondary mm-text--body-md-medium mm-box--margin-top-3 mm-box--padding-0 mm-box--padding-right-4 mm-box--padding-left-4 mm-box--display-inline-flex mm-box--justify-content-center mm-box--align-items-center mm-box--width-full mm-box--color-primary-default mm-box--background-color-transparent mm-box--rounded-pill mm-box--border-color-primary-default box--border-style-solid box--border-width-1"
        data-testid="smart-transaction-status-page-footer-close-button"
      >
        View activity
      </button>
    </div>
  </div>
</div>
`;

exports[`SmartTransactionStatusPage renders the "reverted" STX status 1`] = `
<div>
  <div
    class="mm-box smart-transaction-status-page mm-box--margin-bottom-0 mm-box--display-flex mm-box--flex-direction-column mm-box--align-items-center mm-box--width-full mm-box--height-full mm-box--border-style-none"
  >
    <div
      class="mm-box mm-box--padding-right-4 mm-box--padding-left-4 mm-box--display-flex mm-box--flex-direction-column mm-box--justify-content-center mm-box--align-items-center mm-box--width-full"
      style="flex-grow: 1;"
    >
      <div
        class="mm-box mm-box--padding-right-6 mm-box--padding-left-6 mm-box--display-flex mm-box--flex-direction-column mm-box--align-items-center"
      >
        <div
          class="mm-box mm-box--display-flex"
          style="font-size: 48px;"
        >
          <span
            class="mm-box mm-icon mm-icon--size-inherit mm-box--margin-bottom-4 mm-box--display-inline-block mm-box--color-error-default"
            style="mask-image: url('./images/icons/danger.svg');"
          />
        </div>
        <h4
          class="mm-box mm-text mm-text--heading-md mm-text--font-weight-bold mm-box--color-text-default"
        >
          Your transaction failed
        </h4>
        <div
          class="mm-box mm-box--display-flex mm-box--flex-direction-column mm-box--align-items-center"
        >
          <p
            class="mm-box mm-text mm-text--body-sm mm-box--margin-top-2 mm-box--color-text-alternative"
          >
            Sudden market changes can cause failures. If the problem continues, reach out to MetaMask customer support.
          </p>
        </div>
        <div
          class="mm-box mm-box--margin-top-2 mm-box--display-flex mm-box--flex-direction-column"
        >
          <button
            class="mm-box mm-text mm-button-base mm-button-link mm-button-link--size-auto mm-text--body-md-medium mm-box--padding-0 mm-box--padding-right-0 mm-box--padding-left-0 mm-box--display-inline-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-primary-default mm-box--background-color-transparent"
            type="link"
          >
            View  transaction
          </button>
        </div>
      </div>
    </div>
    <div
      class="mm-box smart-transaction-status-page__footer mm-box--padding-4 mm-box--padding-bottom-0 mm-box--display-flex mm-box--flex-direction-column mm-box--width-full"
    >
      <button
        class="mm-box mm-text mm-button-base mm-button-base--size-md mm-button-secondary mm-text--body-md-medium mm-box--margin-top-3 mm-box--padding-0 mm-box--padding-right-4 mm-box--padding-left-4 mm-box--display-inline-flex mm-box--justify-content-center mm-box--align-items-center mm-box--width-full mm-box--color-primary-default mm-box--background-color-transparent mm-box--rounded-pill mm-box--border-color-primary-default box--border-style-solid box--border-width-1"
        data-testid="smart-transaction-status-page-footer-close-button"
      >
        View activity
      </button>
    </div>
  </div>
</div>
`;

exports[`SmartTransactionStatusPage renders the "success" STX status 1`] = `
<div>
  <div
    class="mm-box smart-transaction-status-page mm-box--margin-bottom-0 mm-box--display-flex mm-box--flex-direction-column mm-box--align-items-center mm-box--width-full mm-box--height-full mm-box--border-style-none"
  >
    <div
      class="mm-box mm-box--padding-right-4 mm-box--padding-left-4 mm-box--display-flex mm-box--flex-direction-column mm-box--justify-content-center mm-box--align-items-center mm-box--width-full"
      style="flex-grow: 1;"
    >
      <div
        class="mm-box mm-box--padding-right-6 mm-box--padding-left-6 mm-box--display-flex mm-box--flex-direction-column mm-box--align-items-center"
      >
        <div
          class="mm-box mm-box--display-flex"
          style="font-size: 48px;"
        >
          <span
            class="mm-box mm-icon mm-icon--size-inherit mm-box--margin-bottom-4 mm-box--display-inline-block mm-box--color-success-default"
            style="mask-image: url('./images/icons/confirmation.svg');"
          />
        </div>
        <h4
          class="mm-box mm-text mm-text--heading-md mm-text--font-weight-bold mm-box--color-text-default"
        >
          Your transaction is complete
        </h4>
        <div
          class="mm-box mm-box--margin-top-2 mm-box--display-flex mm-box--flex-direction-column"
        >
          <button
            class="mm-box mm-text mm-button-base mm-button-link mm-button-link--size-auto mm-text--body-md-medium mm-box--padding-0 mm-box--padding-right-0 mm-box--padding-left-0 mm-box--display-inline-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-primary-default mm-box--background-color-transparent"
            type="link"
          >
            View  transaction
          </button>
        </div>
      </div>
    </div>
    <div
      class="mm-box smart-transaction-status-page__footer mm-box--padding-4 mm-box--padding-bottom-0 mm-box--display-flex mm-box--flex-direction-column mm-box--width-full"
    >
      <button
        class="mm-box mm-text mm-button-base mm-button-base--size-md mm-button-secondary mm-text--body-md-medium mm-box--margin-top-3 mm-box--padding-0 mm-box--padding-right-4 mm-box--padding-left-4 mm-box--display-inline-flex mm-box--justify-content-center mm-box--align-items-center mm-box--width-full mm-box--color-primary-default mm-box--background-color-transparent mm-box--rounded-pill mm-box--border-color-primary-default box--border-style-solid box--border-width-1"
        data-testid="smart-transaction-status-page-footer-close-button"
      >
        View activity
      </button>
    </div>
  </div>
</div>
`;

exports[`SmartTransactionStatusPage renders the "unknown" STX status 1`] = `
<div>
  <div
    class="mm-box smart-transaction-status-page mm-box--margin-bottom-0 mm-box--display-flex mm-box--flex-direction-column mm-box--align-items-center mm-box--width-full mm-box--height-full mm-box--border-style-none"
  >
    <div
      class="mm-box mm-box--padding-right-4 mm-box--padding-left-4 mm-box--display-flex mm-box--flex-direction-column mm-box--justify-content-center mm-box--align-items-center mm-box--width-full"
      style="flex-grow: 1;"
    >
      <div
        class="mm-box mm-box--padding-right-6 mm-box--padding-left-6 mm-box--display-flex mm-box--flex-direction-column mm-box--align-items-center"
      >
        <div
          class="mm-box mm-box--display-flex"
          style="font-size: 48px;"
        >
          <span
            class="mm-box mm-icon mm-icon--size-inherit mm-box--margin-bottom-4 mm-box--display-inline-block mm-box--color-error-default"
            style="mask-image: url('./images/icons/danger.svg');"
          />
        </div>
        <h4
          class="mm-box mm-text mm-text--heading-md mm-text--font-weight-bold mm-box--color-text-default"
        >
          Your transaction failed
        </h4>
        <div
          class="mm-box mm-box--display-flex mm-box--flex-direction-column mm-box--align-items-center"
        >
          <p
            class="mm-box mm-text mm-text--body-sm mm-box--margin-top-2 mm-box--color-text-alternative"
          >
            Sudden market changes can cause failures. If the problem continues, reach out to MetaMask customer support.
          </p>
        </div>
        <div
          class="mm-box mm-box--margin-top-2 mm-box--display-flex mm-box--flex-direction-column"
        >
          <button
            class="mm-box mm-text mm-button-base mm-button-link mm-button-link--size-auto mm-text--body-md-medium mm-box--padding-0 mm-box--padding-right-0 mm-box--padding-left-0 mm-box--display-inline-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-primary-default mm-box--background-color-transparent"
            type="link"
          >
            View  transaction
          </button>
        </div>
      </div>
    </div>
    <div
      class="mm-box smart-transaction-status-page__footer mm-box--padding-4 mm-box--padding-bottom-0 mm-box--display-flex mm-box--flex-direction-column mm-box--width-full"
    >
      <button
        class="mm-box mm-text mm-button-base mm-button-base--size-md mm-button-secondary mm-text--body-md-medium mm-box--margin-top-3 mm-box--padding-0 mm-box--padding-right-4 mm-box--padding-left-4 mm-box--display-inline-flex mm-box--justify-content-center mm-box--align-items-center mm-box--width-full mm-box--color-primary-default mm-box--background-color-transparent mm-box--rounded-pill mm-box--border-color-primary-default box--border-style-solid box--border-width-1"
        data-testid="smart-transaction-status-page-footer-close-button"
      >
        View activity
      </button>
    </div>
  </div>
</div>
`;

exports[`SmartTransactionStatusPage renders the component with initial props 1`] = `
<div>
  <div
    class="mm-box smart-transaction-status-page mm-box--margin-bottom-0 mm-box--display-flex mm-box--flex-direction-column mm-box--align-items-center mm-box--width-full mm-box--height-full mm-box--border-style-none"
  >
    <div
      class="mm-box mm-box--padding-right-4 mm-box--padding-left-4 mm-box--display-flex mm-box--flex-direction-column mm-box--justify-content-center mm-box--align-items-center mm-box--width-full"
      style="flex-grow: 1;"
    >
      <div
        class="mm-box mm-box--padding-right-6 mm-box--padding-left-6 mm-box--display-flex mm-box--flex-direction-column mm-box--align-items-center"
      >
        <div
          class="mm-box mm-box--display-flex"
          style="font-size: 48px;"
        >
          <span
            class="mm-box mm-icon mm-icon--size-inherit mm-box--margin-bottom-4 mm-box--display-inline-block mm-box--color-primary-default"
            style="mask-image: url('./images/icons/clock.svg');"
          />
        </div>
        <h4
          class="mm-box mm-text mm-text--heading-md mm-text--font-weight-bold mm-box--color-text-default"
        >
          Submitting your transaction
        </h4>
        <div
          class="mm-box mm-box--display-flex mm-box--flex-direction-column mm-box--align-items-center mm-box--width-full"
        >
          <div
            class="smart-transaction-status-page__loading-bar-container"
          >
            <div
              class="smart-transaction-status-page__loading-bar"
              style="width: 0%;"
            />
          </div>
        </div>
        <div
          class="mm-box mm-box--display-flex mm-box--flex-direction-column mm-box--align-items-center"
        >
          <p
            class="mm-box mm-text mm-text--body-sm mm-box--margin-top-2 mm-box--color-text-alternative"
          >
            <span>
               
              Estimated completion in &lt; 
              <p
                class="mm-box mm-text smart-transaction-status-page__countdown mm-text--body-sm mm-text--text-align-center mm-box--display-inline-block mm-box--color-text-alternative"
              >
                0:45
              </p>
              
               
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      class="mm-box smart-transaction-status-page__footer mm-box--padding-4 mm-box--padding-bottom-0 mm-box--display-flex mm-box--flex-direction-column mm-box--width-full"
    >
      <button
        class="mm-box mm-text mm-button-base mm-button-base--size-md mm-button-secondary mm-text--body-md-medium mm-box--margin-top-3 mm-box--padding-0 mm-box--padding-right-4 mm-box--padding-left-4 mm-box--display-inline-flex mm-box--justify-content-center mm-box--align-items-center mm-box--width-full mm-box--color-primary-default mm-box--background-color-transparent mm-box--rounded-pill mm-box--border-color-primary-default box--border-style-solid box--border-width-1"
        data-testid="smart-transaction-status-page-footer-close-button"
      >
        View activity
      </button>
    </div>
  </div>
</div>
`;
